charts/wildfly-common/templates/_statefulset.yaml

```diff
@@ -0,0 +1,31 @@
+{{- define "wildfly-common.statefulset.tpl" -}}
+kind: StatefulSet
+apiVersion: apps/v1
+metadata:
+  name: {{ include "wildfly-common.appName" . }}
+  labels: {}
+  {{- if and .Values.build.enabled (eq .Values.build.output.kind "ImageStreamTag") }}
+  annotations:
+    image.openshift.io/triggers: |-
+      [
+        {
+          "from":{
+            "kind":"ImageStreamTag",
+            "name":"{{ include "wildfly-common.appImage" . }}"
+          },
+          "fieldPath":"spec.template.spec.containers[0].image"
+        }
+      ]
+  {{- end }}
+spec:
```

Comment:
Helm Charts creates the StatefulSet before the image is built and available at the OpenShift registry (that's expected), so it is necessary for the StatefulSet to propagate the new image name when it is available to the pod, which will be waiting due to image pull back off. That's a problem here because the StatefullSet propagates the changes only when the pods are ready.

A possible workaround is to change the pod `podManagementPolicy` to `parallel`, indeed that is what we use in the wildfly operator, but that will even start pods in parallel, so I'm sure if that is what you want to have here too and also I am not sure now why the operator uses it since it breaks what a user could be expecting by default from a StatefulSet.

Reply:
mmh, you are right, that's annoying.
 AFAICT, having the `podManagementPolicy` set to `parallel` is fine for the operator because it orchestrates the scaling down of the pods.

For the Helm chart, this would break the exception of a statefulset as you mentioned.

The simplest workaround if the Helm chart build and deploys the image is to start with `replicas: 0` and scale up once the image is build.
@yersan Do you think we should report the issue to OpenShift so that the availability of an image stream tag would trigger a proper deployment of the stateful set?

Reply:
> AFAICT, having the podManagementPolicy set to parallel is fine for the operator because it orchestrates the scaling down of the pods.

Right, I missed that one, although that it looks like that's only applicable at scale down. I'll double-check this in the Operator anyway.